kotlinx-coroutines-core/native/src/MultithreadedDispatchers.kt

```diff
@@ -52,10 +55,24 @@ internal class WorkerDispatcher(name: String) : CloseableCoroutineDispatcher(),
             override fun dispose() {
                 disposableHolder.value = null
             }
+
+            fun isDisposed() = disposableHolder.value == null
+        }
+
+        fun Worker.runAfterDelay(block: DisposableBlock, targetMoment: TimeMark) {
+            if (block.isDisposed()) return
+            val durationUntilTarget = -targetMoment.elapsedNow()
```

Comment:
Was a bit confused at the beginning when looking at documentation only, have filed https://youtrack.jetbrains.com/issue/KT-59044/Improve-various-aspects-of-TimeSource-documentation

Reply:
Probably we can ignore "infinity" problem for now -- such check would be untestable anyway